website/static/js/projectorganizer.js

```diff
@@ -1199,7 +867,470 @@ function _cleanupMithril() {
+                            className : 'text-info'
+                        }, 'Add'),
+                        dismissIcon
+                    ]
+                    )
+                )
+        ];
+        templates.rename = [
+            m('.col-xs-9', m('input#renameInput.tb-header-input', { value : ctrl.items()[0] ? ctrl.items()[0].data.name : '' })),
+            m('.col-xs-3.tb-buttons-col',
+                m('.fangorn-toolbar.pull-right',
+                    [
+                        m.component(Fangorn.Components.button, {
+                            onclick: function () {
+                                _renameEvent.call(ctrl.tb);
+                            },
+                            tooltip: 'Rename collection or project',
+                            icon : 'fa fa-pencil',
+                            className : 'text-info'
+                        }, 'Rename'),
+                        dismissIcon
+                    ]
+                    )
+                )
+        ];
+        templates.addProject = [
+            m('.col-xs-9', [
+                m('input#addprojectInput.tb-header-input', {
+                    config : function () {
+                        applyTypeahead.call(ctrl.tb);
+                    },
+                    type : 'text',
+                    placeholder : 'Name of the project to find'
+                }),
+                m('#add-link-warning.text-warning.p-sm')
+            ]
+                ),
+            m('.col-xs-3.tb-buttons-col',
+                m('.fangorn-toolbar.pull-right',
+                    [
+                        m.component(Fangorn.Components.button, {
+                            onclick: function () {
+                                addProjectEvent.call(ctrl.tb);
+                            },
+                            tooltip: 'Add project',
+                            icon : 'fa fa-plus',
+                            className : 'text-info'
+                        }, 'Add'),
+                        dismissIcon
+                    ]
+                    )
+                )
+        ];
+        generalButtons.push(
+            m.component(Fangorn.Components.button, {
+                onclick: function (event) {
+                    ctrl.mode('search');
+                    ctrl.tb.clearMultiselect();
+                },
+                tooltip: 'Filter visible items',
+                icon: 'fa fa-search',
+                className : 'text-primary'
+            }, 'Search'),
+            m.component(Fangorn.Components.button, {
+                onclick: function (event) {
+                    showLegend.call(ctrl.tb);
+                },
+                tooltip: 'Learn more about how to use the project organizer.',
+                icon: 'fa fa-info',
+                className : 'text-info'
+            }, '')
+        );
+        if (ctrl.items().length === 1) {
+            rowButtons = _poDefineToolbar.call(ctrl.tb, ctrl.items()[0]);
```

Comment:
Why not make `_poDefineToolbar` into a component as well? Perhaps call it `POToolbarButtons`, so you could do:

```js
rowButtons = m.component(POToolbarButtons, {treebeard: ctrl.tb, item: ctrl.items()[0]});
```